crates/bevy_text/src/text.rs

```diff
@@ -279,6 +279,21 @@ pub enum FontSmoothing {
     // SubpixelAntiAliased,
 }
 
+/// Provides convenience methods for constructing text blocks.
+pub trait TextBuilderExt {
+    /// Adds a flat list of spans as children of the current entity.
+    fn with_spans<S: Component>(&mut self, spans: Vec<(S, TextStyle)>) -> &mut Self;
```

Comment:
It might be more ergnomic if the `spans` parameter is `impl IntoIterator<Item = (String, TextStyle)>`, and the string gets wrapped automatically.

Reply:
`impl IntoIterator<Item = (impl Into<String>, TextStyle)>` even.

Reply:
Fixed in https://github.com/bevyengine/bevy/pull/15591/commits/a12f0a6fac29e00b37fdd0299130fd3117b3333e